### PR TITLE
sign image and upload sbom

### PR DESCRIPTION
**Pull request by sunnyyip**

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -15,16 +15,24 @@
 name: release-guac-image
 
 on:
-  release:
-    types: [published]
+  push:
+    tags:
+      - 'v*'
```
Comment (jeffmendoza) [inline .github/workflows/release.yaml:20]: Do we want to create releases manually or have the releases created by pushing tags?

Comment (sunnyyip) [reply .github/workflows/release.yaml:20]: It looks like a tag is a pre-req of creating a release. I changed the trigger of the image build workflow to react to tag push. So we could potentially test the image prior to cutting a release or build beta images that is not intended for a release.

We could add logic to auto-generate the release if we want. Right now we can still cut the release manually and have the image tag matches it.

Comment (jeffmendoza) [reply .github/workflows/release.yaml:20]: Let's add a doc (release.md / developer.md ?) with our intended workflow:

1. Push tag
2. verify images
3. Create release in GitHub

... etc.

Comment (sunnyyip) [reply .github/workflows/release.yaml:20]: Yup, makes sense and we should make a decision on how we want to do 3.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -34,16 +42,39 @@ jobs:
...
       - name: Build and publish image
+        id: build_n_publish_image 
         run: |
           #!/usr/bin/env bash
           set -euo pipefail
-          pack build ${IMAGE}:${GITHUB_REF_NAME} --builder ${BUILDER} --buildpack ${BUILDPACK} --publish
+          VERSION=$(git describe --tags --always)
+          COMMIT=$(git rev-parse HEAD)
+          BUILD=$(date +%FT%T%z)
+          PKG=github.com/${{ github.repository }}/pkg/version
```
Comment (jeffmendoza) [inline .github/workflows/release.yaml:59]: This needs to match the package name in the [go.mod](https://github.com/guacsec/guac/blob/main/go.mod#L1). If someone forks it and tries to make a release, it won't get the verison/commit/date.

**Review by jeffmendoza — APPROVED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (jeffmendoza) [reply .github/workflows/release.yaml:20]: Added #791

File: .github/workflows/release.yaml (hunk 2 above)
Comment (sunnyyip) [reply .github/workflows/release.yaml:59]: good point...I'll put the repo name back

**Pull request closed without merging**